### ISSUE TITLE
Build fails at the beginning

### ISSUE THREAD

**Issue by ghost** [peters/openalpr-windows#4]
**My build tools**:
- Visual Studio 2010
- PowerShell 2.0

**Command to build the suit**:
- .\build.ps1 -Configuration Debug -Platform Win32 -PlatformToolset v100 CudaGeneration None

**Output**:

<pre>PlatformToolset: v100

C:\Users\admin\AppData\Local\Temp\tmpF9C3.tmp.cmd
Join-Path : Cannot bind argument to parameter 'Path' because it is null.
At F:\reza\alpr\openalpr\windows\build.ps1:243 char:32
+     $script:MsbuildExe = Join-Path <<<<  (${Env:ProgramFiles(x86)}) MSBuild\$ToolsVersion\Bin\msbuild.exe
    + CategoryInfo          : InvalidData: (:) [Join-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.JoinPathCom
   mand

PlatformToolset: Successfully configured msvs PlatformToolset v100

Test-Path : Cannot bind argument to parameter 'Path' because it is null.
At F:\reza\alpr\openalpr\windows\build.ps1:150 char:20
+     if(-not (Test-Path <<<<  $MsbuildExe)) {
    + CategoryInfo          : InvalidData: (:) [Test-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.TestPathCom
   mand

Tesseract: Debug, Win32, v100

Msbuild: Project - F:\reza\alpr\openalpr\windows\tesseract-ocr\dependencies\giflib\giflib.vcxproj

Msbuild: OutDir - F:\reza\alpr\openalpr\windows\build\artifacts\2.1.0\v100\Debug\Win32\tesseract

Msbuild: Matrix - Debug, Win32, v100

Exception calling "Start" with "0" argument(s): "Cannot start process because a file name has not been provided."
At F:\reza\alpr\openalpr\windows\build.ps1:184 char:19
+     $Process.Start <<<< () | Out-Null
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : DotNetMethodException

Exception calling "WaitForExit" with "0" argument(s): "No process is associated with this object."
At F:\reza\alpr\openalpr\windows\build.ps1:185 char:25
+     $Process.WaitForExit <<<< ()
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : DotNetMethodException

Die :  returned a non-zero exit code
At F:\reza\alpr\openalpr\windows\build.ps1:188 char:12
+         Die <<<<  ("{0} returned a non-zero exit code" -f $Filename)
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Die</pre>

Any thoughts?

**Comment by juampitraver** [comment]
I have a similar issue

PlatformToolset: Successfully configured msvs PlatformToolset v120

Tesseract: Debug, x64, v120

Msbuild: Project - C:\Windows\system32\openalpr\windows\tesseract-ocr\dependencies\giflib\giflib.vcxproj

Msbuild: OutDir - C:\Windows\system32\openalpr\windows\build\artifacts\2.1.0\v120\Debug\x64\tesseract

Msbuild: Matrix - Debug, x64, v120

Microsoft (R) Build Engine version 12.0.31101.0
[Microsoft .NET Framework, version 4.0.30319.34014]
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: C:\Windows\system32\openalpr\windows\tesseract-ocr\dependencies\giflib\giflib.vcxproj

Die : C:\Program Files (x86)\MSBuild\12.0\Bin\msbuild.exe returned a non-zero exit code
At C:\Windows\system32\openalpr\windows\build.ps1:194 char:9
-         Die ("{0} returned a non-zero exit code" -f $Filename)
-         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
  - FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Die

any solution??

thanks

**Comment by ghost** [comment]
The build script assumes that your OS is 64-bit. My Windows was 32-bit and therefore there was no `Program Files (x86)` directory. So I just removed the `(x86)` from the build script file and went on.

**Comment by juampitraver** [comment]
My OS is 64-bit and there is a "C:\Program Files (x86)\MSBuild\12.0\Bin\msbuild.exe"
other solution??

thanks

**Comment by aybassiouny** [comment]
Removing the "(x86)" from line 249 fixed this issue for me.

**Comment by seekerquoc** [comment]
i have same problem, my msbuild.exe aslo x86 but still show error

**Comment by aybassiouny** [comment]
@seekerquoc did you try to remove the "(x86)" from line 249 of build.ps1?

**Comment by seekerquoc** [comment]
no, i dont remove, problem with sln libteressact, i fixed it, build succesful.

Thanks

**Comment by tinarsoft** [comment]
this error is for openaplr-windows dependency subProject

please download or clone :
1-opencv   (in root folder)
2-tesseract   (in 'tesseract-ocr/tesseract' folder)
